kubernetes-itests/src/test/java/io/fabric8/kubernetes/ConfigMapIT.java

```diff
@@ -101,14 +114,26 @@ public void update() {
 
   @Test
   public void delete() {
+    Ready<ConfigMap> configMap1Ready = new Ready<ConfigMap>(ConfigMap.class, client, "configmap1", currentNamespace);
+    await().atMost(30, TimeUnit.SECONDS).until(configMap1Ready);
     assertTrue(client.configMaps().inNamespace(currentNamespace).withName("configmap1").delete());
+
+    Ready<ConfigMap> configMap2Ready = new Ready<ConfigMap>(ConfigMap.class, client, "configmap2", currentNamespace);
+    await().atMost(30, TimeUnit.SECONDS).until(configMap2Ready);
     assertTrue(client.configMaps().inNamespace(currentNamespace).withName("configmap2").delete());
   }
 
   @After
   public void cleanup() throws InterruptedException {
-    client.configMaps().inNamespace(currentNamespace).delete();
-    // Wait for resources to get destroyed
-    Thread.sleep(30000);
+    try {
+      if (client.configMaps().inNamespace(currentNamespace).list().getItems().size()!= 0) {
+        client.configMaps().inNamespace(currentNamespace).delete();
+      }
+
+      Delete<ConfigMap> configMapDelete = new Delete<ConfigMap>(ConfigMap.class, client, "configmap1", currentNamespace);
+      await().atMost(30, TimeUnit.SECONDS).until(configMapDelete);
+    } catch (NullPointerException exception) {
+      cleanup();
```

Comment:
Isn't it a potentially endless recursion here? Why are we handling `NPE` here?

Reply:
I followed this up from [here](https://github.com/fabric8io/kubernetes-client/pull/1121/files#diff-e892e898797ea5223e863136b7348d7cR134) .

If the 'entity/resource' is not responding yet, we can catch the exception and try cleaning up again. I think, practically, the recursive loop here being endless is unlikely. Therefore, I let it there as good measure.

@rohanKanojia what do you think?

Reply:
Even if ending up in a recursive loop is unlikely; I believe we should limit this to some finite number of retry attempts, rather than having no control over the number of retries encountered